### PR TITLE
Compute non-large NTTs in CPU backend without parallelism

### PR DESCRIPTION
**Pull request by ShanieWinitz**
## Describe the changes

This PR...

## Linked Issues

Resolves #

### CODE REVIEW

File: icicle/backend/cpu/include/ntt_tasks_manager.h [hunk 1]
```diff
@@ -86,6 +86,7 @@ namespace ntt_cpu {
     uint32_t nof_pending_tasks = 0; // the current count of tasks that are pending execution
 
   private:
+    const uint32_t logn;                             // log of the NTT size
```
Comment (mickeyasa) [inline icicle/backend/cpu/include/ntt_tasks_manager.h:89]: I didn't fully understand the logn change.

File: icicle/backend/cpu/include/ntt_task.h [hunk 2]
```diff
@@ -72,11 +73,15 @@ namespace ntt_cpu {
   template <typename S, typename E>
   void NttTask<S, E>::execute()
   {
-    if (!ntt_task_coordinates->reorder) {
+    if (!ntt_data->is_parallel) {
       hierarchy_0_cpu_ntt();
     } else {
```
Comment (mickeyasa) [inline icicle/backend/cpu/include/ntt_task.h:78]: a simple || can replace this complex if else

**Review by mickeyasa — APPROVED** [review]
great

File: icicle/backend/cpu/include/ntt_tasks_manager.h (hunk 1 above)
Comment (ShanieWinitz) [reply icicle/backend/cpu/include/ntt_tasks_manager.h:89]: Discussed this in person

File: icicle/backend/cpu/include/ntt_utils.h [hunk 3]
```diff
@@ -47,4 +47,25 @@ namespace ntt_cpu {
...
+   * @param log_batch_size The log of the batch size.
+   * @param scalar_size The size of the scalars.
+   * @return bool Returns true if the NTT computation should be parallelized, or false otherwise.
+   */
+  bool is_parallel(uint32_t log_size, uint32_t log_batch_size, uint32_t scalar_size)
+  {
+    // for small scalars, the threshold for when it is faster to use parallel NTT is higher
+    if (
+      (scalar_size >= 32 && (log_size + log_batch_size) <= 13) ||
+      (scalar_size < 32 && (log_size + log_batch_size) <= 16)) {
```
Comment (Koren-Brand) [inline icicle/backend/cpu/include/ntt_utils.h:66]: How did you get these specific numbers?

Comment (ShanieWinitz) [reply icicle/backend/cpu/include/ntt_utils.h:66]: I ran tests with all sizes on my Mac and the 'Miner' machine, both in parallel and serial modes. Based on the results, I selected the best threshold depending on the scalar's size.

**Review by Koren-Brand — APPROVED** [review]

**Pull request merged**